Spawn-App-iOS-SwiftUI/ViewModels/ChoosingTags/ChoosingTagViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  ChoosingTagViewModel.swift
+//  Spawn-App-iOS-SwiftUI
+//
+//  Created by Michael Tham on 23/1/25.
+//
+
+import Foundation
+
+class ChooseTagPopUpViewModel: ObservableObject {
+	@Published var chooseTagErrorMessage: String = ""
+    @Published var tags: [UUID] = []
+    @Published var selectedTags: Set<UUID> = []
+    
+    var userId: UUID
+
+    var apiService: IAPIService = MockAPIService()
+
+    init(userId: UUID, apiService: IAPIService) {
+        self.userId = userId
+		self.apiService = apiService
+	}
+
+	func AddTagsToFriend(friendUserId: UUID, friendTagIds: [UUID]) async {
+		if let url = URL(
+			string: APIService.baseURL + "friendTags/addUserToTags/\(userId)")
+		{
+			do {
+				try await self.apiService.sendData(
+					friendTagIds,
+					to: url,
+					parameters: [
+						"friendUserId": friendUserId.uuidString
+					])
+			} catch {
+				await MainActor.run {
+					chooseTagErrorMessage =
+						"There was an error select a tag for your friend. Please try again."
+					print(apiService.errorMessage ?? "")
+				}
+			}
+		}
+	}
+    
+    func fetchTagsToAddToFriend(friendUserId: UUID) async {
+        let urlString = APIService.baseURL + "friendTags/addUserToTags/\(userId)"
+        
+        if let url = URL(string: urlString) {
+            let parameters: [String: String] = [
+                "friendUserId": friendUserId.uuidString
+            ]
+        
+            do {
+                let fetchedTags: [FriendTag] = try await self.apiService.fetchData(
+                    from: url, parameters: parameters
+                )
+
+                await MainActor.run {
+                    self.tags = fetchedTags.map { $0.id }
```

Comment:
```suggestion
                    self.tags = fetchedTags
```
This can probably be changed as such, after editing the var type above